row.py

```diff
@@ -81,9 +100,215 @@ def convert_pdf_to_pil(pdf_as_bytes):
     try:
         images = convert_from_bytes(pdf_as_bytes, dpi)
     except (TypeError, PDFInfoNotInstalledError, PDFPageCountError, PDFSyntaxError) as error:
-        print(error)
+        logging.debug(error)
         messages = error
 
     count = len(images)
 
     return (images, count, messages)
+
+
+def get_circles(item_path, output_path):
+    """detect circles in an image and export them to output_path
+
+    Args:
+        item_path (Path): The location of the image to detect circles in
+        output_path (Path): The output directory for cropped images of detected circles to be stored
+
+    Returns:
+        array: An array of binary images
+    """
+    if not item_path.name.casefold().endswith(("jpg", "jpeg", "tif", "tiff", "png")):
+
+        return []
+
+    logging.debug("detecting circles in %s", item_path)
+    #: read in image from bytes
+    byte_img = item_path.read_bytes()
+    img = cv2.imdecode(np.frombuffer(byte_img, dtype=np.uint8), 1)  # 1 means flags=cv2.IMREAD_COLOR
+
+    if output_path:
```

Comment:
We can probably get rid of L130-134.  This makes a copy of the original image in the output folder, which I mainly did to make sure it looked correct when I read it in using the read_bytes/cv2.imdecode/np.frombuffer method.